### DISCUSSION TITLE
How to install w/ language support via `nix`?

### DISCUSSION THREAD

**Discussion by nathan-at-least** [helix-editor/helix#7067] [Q&A]
Hi, I run debian with `nix` installed.

So far I've been using `hx` built from source, and I've installed various LSP supports in an ad-hoc manner (without nix), and I'm regretting this. Not because of `hg` or different language plugins, they're awesome once configured! Instead it is because each LSP server uses various different implementation-specific packaging, so my "fully working `hx` configuration" is very non-reproducible and will take a lot of effort to set up on other systems. This is exactly the kind of complexity `nix` is designed to accommodate!

Is there a way I can specify "I want `hx` along with language support for X, Y, and Z" somehow, then rely on `nix` to set all that up correctly so that `hx --health X` succeeds all checks for all the languages I want? I think I'd also be happy with a `nix` install that simply includes all possible LSP providers for as many languages as is feasible.

The [nixos install section](https://docs.helix-editor.com/install.html#nixos) doesn't mention anything about language support.

I've skimmed over `flake.nix` but I don't see anything obviously about LSP support for various languages.

**Comment by the-mikedavis** [comment]
I don't think we're intested in providing something like this officially but you could create a Nix library that maps the languages to the language server binaries and wraps the `hx` binary in a script that set `PATH` to include the desired language server binaries in the Nix store.

**Comment by MatrixManAtYrService** [comment]
It does seem like it would be pretty easy to cobble together a nixpkgs package which does nothing except depend on the various language servers that helix uses.  However it's also been pretty easy to just add them to a home-manager module and then reference that in my system configuration flake.

I haven't done all of them, but so far they've been incredibly easy to find.  Their names are either:
- the string shown in `hx --health`
- `nodePackages_latest.` + the string shown in `hx --health`
